bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/handler/EvohomeAccountBridgeHandler.java

```diff
@@ -236,15 +262,33 @@ private void updateThings() {
         Map<String, String> zoneIdToTcsIdMap = new HashMap<>();
         Map<String, ThingStatus> idToTcsThingsStatusMap = new HashMap<>();
 
+        EvohomeApiClient localApiClient = apiClient;
+        if (localApiClient == null) {
+            return;
+        }
         // First, create a lookup table
-        for (LocationStatus location : apiClient.getInstallationStatus()) {
+        LocationsStatus localLocationsStatus = localApiClient.getInstallationStatus();
+        if (localLocationsStatus == null) {
+            return;
+        }
+        for (LocationStatus location : localLocationsStatus) {
             for (GatewayStatus gateway : location.getGateways()) {
+                if (gateway == null) {
+                    continue;
+                }
                 for (TemperatureControlSystemStatus tcs : gateway.getTemperatureControlSystems()) {
-                    idToTcsMap.put(tcs.getSystemId(), tcs);
-                    tcsIdToGatewayMap.put(tcs.getSystemId(), gateway);
+                    String systemId = tcs.getSystemId();
+                    if (systemId == null) {
+                        continue;
```

Comment:
I guess it could still be relevant to iterate `tcs.getZones()` and populate `idToZoneMap`?

Reply:
I partly reverted the commits that fixed the nesting review comments as otherwise the `zoneHandler.update` and `tcsHandler.update` are not called. As these methods have functionality to set the things offline etc when the passed arguments are null. It would change the functionality of the binding and that is beyond the scope of this PR.

Reply:
The old code populated `idToZoneMap` even if `tcs.getSystemId()` would return null, so there is still a (probably undesired) change in behavior.

Reply:
Might be true, changed it so that idToZoneMap should always be populated